popde/adaptive_kde.py

```diff
@@ -224,6 +224,143 @@ def optimize_parameters(self, method='loo_cv', fom_plot_name=None):
+            train_data, test_data = self.kde_data[train_index], self.kde_data[test_index]
+            local_weights = None # FIX ME
+            awkde = AdaptiveBwKDE(train_data, local_weights, input_transf=self.input_transf,
+                                  stdize=self.stdize, rescale=rescale_factor,
+                                  bandwidth=self.bandwidth, alpha=alpha)
+            fom += np.log(awkde.evaluate(test_data))
+        return -fom
+
+    def kfold_cv_score(self, rescale_factors_alpha, seed=42):
+        """
+        Perform k-fold cross-validation
+        """
+        alpha= rescale_factors_alpha[-1]
+        rescale_factor = rescale_factors_alpha[0:len(rescale_factors_alpha)-1]
+        from sklearn.model_selection import KFold
+        kf = KFold(n_splits=self.n_splits, shuffle=True, random_state=seed)
+        fom = []
+        for train_index, test_index in kf.split(self.kde_data):
+            train_data, test_data = self.kde_data[train_index], self.kde_data[test_index]
+            local_weights = None # FIX ME
+            awkde = AdaptiveBwKDE(train_data, local_weights, input_transf=self.input_transf,
+                                  stdize=self.stdize, rescale=rescale_factor,
+                                  bandwidth=self.bandwidth, alpha=alpha)
+            log_kde_eval = np.log(awkde.evaluate(test_data))
+            fom.append(log_kde_eval.sum())
+        return -sum(fom)
+
+    def optimize_rescale_parameters(self, initial_rescale_factor, initial_alpha=0.0, method='loo_cv', fom_plot_name=None, bounds=None):
+
+        """
+        Fixed bounds on rescale factor otehrwise it is not working
+        or use better initial guesses using MultiRescaling class
+        """
+
+        initial_choices = np.concatenate((initial_rescale_factor, initial_alpha))
+        from scipy.optimize import minimize
+        best_params = {'rescale_per_dim': None}
+        # Perform Nelder-mead based Optimization
+        if method == 'kfold_cv':
+            result = minimize(
+                    self.kfold_cv_score,        # func to minimize why negative?
+                    initial_choices,             # Initial guess for the parameters
+                    # args= ( )  #Additional arguments to pass to the objective function
+                    method='Nelder-Mead',      # Optimization method
+                    options={'disp': True}     # Display optimization progress
+                    , bounds=bounds
+            )
+        else:
+            result = minimize(
+                    self.loo_cv_score,        # why negative
+                    initial_choices,     
+                    #args=(),  # Additional arguments to pass to the objective function
+                    method='Nelder-Mead',
+                    options={'disp': True} #, bounds #crucial maybe
+            )
+
+        #make self. rescale be the optimized results
+        self.rescale = result.x 
```

Comment:
I think you need to set alpha as well no?  If alpha is one of the parameters in the optimization you will have Ndim + 1 outputs for 'result.x'